### PR TITLE
Fix/dashboard user table

### PR DESCRIPTION
**Pull request by Saif-Alhaider**

### CODE REVIEW

**Review by mosayed01 — APPROVED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt [hunk 1]
```diff
@@ -4,12 +4,109 @@ package org.thechance.common.data.remote.gateway
...
+                email = "jane_davis@example.com",
+                country = "Other",
+                permissions = listOf(
+                    UserDto.PermissionDto(id = 1, permission = "END_USER"),
+                    UserDto.PermissionDto(id = 2, permission = "DRIVER")
+                )
+            ),
+        ).toEntity()
+        val startIndex = (page - 1) * numberOfUsers
+        val endIndex = startIndex + numberOfUsers
```
Comment (mujtaba912) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt:94]: endIndex is used in only one place you could remove it and uses it directly in result

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/getway/IRemoteGateway.kt [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+import org.thechance.common.domain.entity.User
+import org.thechance.common.domain.entity.UserTokens
 
 
 interface IRemoteGateway {
 
     fun getUserData(): Admin
 
-    fun getUsers(): List<User>
+    fun getUsers(page: Int, numberOfUsers: Int): DataWrapper<User>
```
Comment (mujtaba912) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/getway/IRemoteGateway.kt:16]: make getUsers a suspend fun so it does not run on main thread

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/GetUsersUseCase.kt [hunk 3]
```diff
@@ -1,14 +1,15 @@
 package org.thechance.common.domain.usecase
 
+import org.thechance.common.data.remote.gateway.FakeRemoteGateway
+import org.thechance.common.domain.entity.DataWrapper
 import org.thechance.common.domain.entity.User
-import org.thechance.common.domain.getway.IRemoteGateway
 
 interface IGetUsersUseCase {
-    operator fun invoke(): List<User>
+    operator fun invoke(page:Int,numberOfUsers:Int): DataWrapper<User>
```
Comment (mujtaba912) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/GetUsersUseCase.kt:8]: make invoke a suspend fun so it does not run on main thread

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt (hunk 1 above)
Comment (Saif-Alhaider) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt:94]: it makes the code cleaner and more readable

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/scaffold/DashboardSideBar.kt [hunk 4]
```diff
@@ -81,7 +81,7 @@ fun DashboardSideBar(
                 )
             )
             //endregion
-            Spacer(Modifier.fillMaxHeight(.1f))
+            Spacer(Modifier.height(40.dp))
```
Comment (Asiasama710) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/composables/scaffold/DashboardSideBar.kt:84]: use spacer for just weight

**Review by mujtaba912 — APPROVED** [review]

**Pull request merged**